proteus/MeshTools.py

```diff
@@ -3734,6 +3740,29 @@ def computeGeometricInfo(self):
         for m in self.meshList:
             m.computeGeometricInfo()
 
+def buildReferenceSimplex(nd=2):
+    """
+    This function creates and returns a Proteus mesh object for the reference 
+    simplex.  This function can helpful for test development.
+    """
+    from proteus import Domain
+    from proteus import TriangleTools
+    if nd == 2:
+        unit_simplex_domain_2d = Domain.unitSimplex(2)
+        polyfile = "reference_triangle"
+        unit_simplex_domain_2d.writePoly(polyfile)
+        tmesh = TriangleTools.TriangleBaseMesh(baseFlags="Yp",
+                                               nbase=1,
+                                               verbose=False)
+        tmesh.readFromPolyFile(polyfile)
+        mesh = tmesh.convertToProteusMesh(verbose=0)
+        mesh.partitionMesh()
+        mesh.globalMesh = mesh
+        return mesh
+    if nd == 3:
+        pass
```

Comment:
Okay, I've added the 3D case, but the updates do not appear to be rendering here after I pushed up the changes? 

Anyway, allow me to summarize what I've done, and some questions I have

1. I'm not entirely sure what I should be returning in the 1D case?  Just a line 0-1?
2. I've updated the old unit_mesh_tests to the new py.test format.
3. I've added the 3D simplex mesh.
4. So I found the code duplication for the tetgen in the NumericalSolution module with the MeshTools unnecessary.  Thus, I wrote a routine in the TriangleTools module that generates a tetrahedral mesh from a polyfile which I call in the nd==3 case.  I'm interested in what you think about (a) using this function call in NumericalSolution and (b) whether TriangleTools is really the correct home for the function.